### ISSUE TITLE
Identity update errors on firefox

### ISSUE THREAD

**Issue by fernandoabolafio** [decred/politeiagui#1249]
A user reported that he cannot update his identity using firefox. As he described:

> yeah, I click the email link and then the message on PI goes away but then as soon as I navigate to a proposal or refresh the page the error msg comes back

I was able to reproduce that for production proposals.decred.org but on Firefox but I could not for localhost.

Another user also reported receiving the error: "Error generating new identity" without further information. This error needs to be more informative if it's either coming from an API cause or something related to browser storage.

**Comment by lukebp** [comment]
Additional comment from ragnarok on matrix:

"and when I reset my identity, it works until I click around a bit. I just tried to re-enter my new ID with the new private/public keys in the “Import Identity” modal, but then it just keeps saying `Error importing identity - The provided public key doesn’t match the key stored in the server`"

Some screenshots of the error messages can be found here:
https://imgur.com/a/XMPCz6l

**Comment by lukebp** [comment]
Some more info:

"I've been using Chrome but I also tried FF as well. I tried incognito with Chrome and a freshly installed FF and both came with the same result and I use Windows 10 64-bit"

**Comment by fernandoabolafio** [comment]
I could reproduce this error on my local setup. I don't know how I got my user in this state but what was happening was that every time I requested to update the user identity, it was being updated to the previously requested update. Something like:

- Update to identity B  results in an update to identity A
- Update to identity C results in an update to identity B

Making the identity in the client and in the server always different. 

After doing some debugging in the server side I realized that my user had two identities activated and every time I requested to update the identity it would add the new identity into the user identities map resulting in 3 active identities, after verifying the update of the last key, just the first index of the 3 would be deactivated, resulting again in two identities updated, since the old one is the first to be found as "active" in the map, the server was always off by one when asked what was the active user identity.
This PR solves this problem: https://github.com/decred/politeia/pull/905. 

However, I could not find what has caused the server to allow a user to have multiple active identities.
